routers/web/repo/issue.go

```diff
@@ -843,12 +843,18 @@ func NewIssue(ctx *context.Context) {
 func NewIssueChooseTemplate(ctx *context.Context) {
 	ctx.Data["Title"] = ctx.Tr("repo.issues.new")
 	ctx.Data["PageIsIssueList"] = true
-	ctx.Data["milestone"] = ctx.FormInt64("milestone")
 
 	issueTemplates := ctx.IssueTemplatesFromDefaultBranch()
-	ctx.Data["NewIssueChooseTemplate"] = len(issueTemplates) > 0
 	ctx.Data["IssueTemplates"] = issueTemplates
 
+	if len(issueTemplates) == 0 {
+		ctx.Redirect(fmt.Sprintf("%s/issues/new?%s", ctx.Repo.Repository.HTMLURL(), ctx.Req.URL.RawQuery), http.StatusSeeOther)
```

Comment:
Is that route correct? Shouldn't this be something like `%s/issues/new?redirect=%s` instead?
(I don't know how that query parameter is actually called)

Reply:
I have tested, it is correct. They share the same parameters `milestone` and `project`.

Reply:
Ah, I think I misunderstood `RawQuery`: I interpreted it as the whole URL.
But it is only meant as the part after the `?`. Then it makes sense of course.

Reply:
I think we could add a comment here why it works

Reply:
Added, it is clear now?